src/sink/udp.rs

```diff
@@ -128,19 +157,21 @@ impl Sink for Udp {
         _reply_channel: Sender<sink::Reply>,
     ) -> Result<()> {
         self.postprocessors = make_postprocessors(processors.post)?;
-        let socket = UdpSocket::bind((self.config.host.as_str(), self.config.port)).await?;
+        let socket =
+            UdpSocket::bind((self.config.bind.host.as_str(), self.config.bind.port)).await?;
         socket
-            .connect((self.config.dst_host.as_str(), self.config.dst_port))
+            .connect((self.config.host.as_str(), self.config.port))
```

Comment:
this should also be guarded by `self.config.bound` as it is done in `on_signal`, no?

Reply:
Good catch! you're spot on :D

Reply:
extracted the whole bind logic in a function that gets called in both places :+1: